src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -1270,22 +1288,36 @@ def should_update_existing_container(self, c):
         return c["name"].lower() == self.get_argument_container_name().lower() or self.get_argument_force_single_container_updates()
 
     def set_up_runtime(self):
-        if self.get_argument_runtime() is not None or self.get_argument_enable_java_metrics() is not None:
-            runtime_option = infer_runtime_option(self.get_argument_runtime(), self.get_argument_enable_java_metrics())
+        if self.should_set_up_runtime():
+            runtime_option = infer_runtime_option(self.get_argument_runtime(), self.get_argument_enable_java_metrics(), self.get_argument_enable_java_agent())
             runtime_def = None  # default value for runtime_option == RUNTIME_GENERIC, set None to erase runtime info
             if runtime_option == RUNTIME_JAVA:
-                if self.get_argument_enable_java_metrics() is None:
-                    runtime_def = {
-                        "java": {}  # empty to keep or setup default java runtime
-                    }
-                else:
-                    runtime_java_def = RuntimeJavaModel
+                runtime_java_def = {}
+                if self.get_argument_enable_java_metrics() is not None:
                     runtime_java_def["enableMetrics"] = self.get_argument_enable_java_metrics()
-                    runtime_def = {
-                        "java": runtime_java_def
-                    }
+
+                if self.get_argument_enable_java_agent() is not None:
+                    runtime_java_def["javaAgent"] = {}
+                    runtime_java_def["javaAgent"]["enabled"] = self.get_argument_enable_java_agent()
+                    if self.get_argument_enable_java_agent() == False:
+                        runtime_java_def["javaAgent"]["logging"] = { # clear logger settings when disable java agent
```

Comment:
```suggestion
                        runtime_java_def["javaAgent"]["logging"] = {  # clear logger settings when disable java agent
```
![image](https://github.com/Azure/azure-cli-extensions/assets/46143499/a068c0ee-83df-4cab-913b-606b19eba4bb)